ballerina/main.bal

```diff
@@ -124,9 +116,18 @@ isolated function parseResponseAsJson(string resp) returns json|error {
     return result;
 }
 
-isolated function parseResponseAsType(string resp, typedesc<anydata> expectedResponseTypedesc) returns anydata|error {
-    json respJson = check parseResponseAsJson(resp);
-    anydata|error result = trap respJson.fromJsonWithType(expectedResponseTypedesc);
+isolated function parseResponseAsType(string resp, 
+            typedesc<anydata> expectedResponseTypedesc, boolean isOriginallyJsonObject) returns anydata|error {
+    if !isOriginallyJsonObject {
+        map<json> respContent = check resp.fromJsonStringWithType();
+        anydata|error result = trap respContent[OBJECT_KEY].fromJsonWithType(expectedResponseTypedesc);
```

Comment:
Add a test for this inline with our discussion last week. One for which ensureType will fail, but fromJsonWithType will result in the correct value.

Reply:
The tests contains both `isOriginallyJsonObject=true` and `isOriginallyJsonObject=false` scenarios